From 3d206da720433ae1a78164a8b7ad4f0a3edc061d Mon Sep 17 00:00:00 2001
From: Rainer Gerhards <rgerhards@adiscon.com>
Date: Mon, 2 Mar 2020 14:52:18 +0100
Subject: [PATCH] string rulebase bugfix: segfault when using LF in json rule

If a json rule used a LF inside a string rule (one not loaded
rulebase file), liblognorm segfaults.

closes https://github.com/rsyslog/liblognorm/issues/333
---
 src/samp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/samp.c b/src/samp.c
index fc00baf..d5d42c3 100644
--- a/src/samp.c
+++ b/src/samp.c
@@ -1019,7 +1019,7 @@ ln_sampRead(ln_ctx ctx, FILE *const __restrict__ repo, const char **inpbuf,
 				done = 1; /* last line missing LF, still process it! */
 		} else if(c == '\n') {
 			++ctx->conf_ln_nbr;
-			if(inParser) {
+			if(inParser && repo != NULL) {
 				if(ln_sampChkRunawayRule(ctx, repo, inpbuf)) {
 					/* ignore previous rule */
 					inParser = 0;
-- 
2.21.1

